Various fixes for slingshot

rcannood opened the pull request:
Hello Kelly,

A while back I made a few fixes on our fork of slingshot. Here is a pull request to integrate these fixes into your slingshot repository.

Kind regards,
Robrecht


**List of changes**
.travis.yml:
* Add apt package for installing splatter.
* Treat warnings as errors! This will cause Travis to fail building, but fixing the warnings is important.
* Use devtools to install suggested packages on Travis.

DESCRIPTION:
* Set minimum required R version back to 3.4.
* Add splatter to imports.

Many R/*.R files:
* Specify princurve package when linking to principal.curve.

R/AllClasses.R:
* Add documentation for curveWeights and pseudotime slots.

R/AllGenerics.R:
* Add description tag, roxygen would otherwise complain.
* Remove old plot3d code.

R/getCurves.R and R/slingshot.R:
* Use hyphen "--" instead of minus sign "-".

R/plotting.R: 
* Fix typo princiapl to principal.

drisso commented:
Hi @rcannood ,

thanks for the pull request!

There are some changes that we cannot accept. Namely:

- Set minimum required R version back to 3.4: as I said elsewhere, R 3.5 is required per Bioconductor policy.
- Add splatter to imports: I'm not sure this is appropriate, as no functionality of slingshot depends on splatter. I think that Suggests: is the right choice here. Moreover, splatter is on Bioconductor so there is no need for remotes here.
- "Visualizing" to "Visualising": we are using American spelling, so visualizing is correct.
- I cannot comment on the plot3d, but @kstreet13 might know if this is indeed old code or if it's used / useful.
- Add documentation for curveWeights and pseudotime slots: there are no curveWeights and pseudo time slots in the class.

I think all the other edits are useful (thanks!), but you will need to create a new pull request with just the changes that are not part of the above list and make sure it passes all checks, if you want us to merge it to master.

As an additional comment, we are in the middle of a submission to Bioconductor, so we would need to wait until that's done before merging your (updated) pull request.

Best,
Davide

rcannood commented:
Hello Davide,

Thanks for the quick reply! I just made a few commits in response to your remarks. 

Robrecht

---------------------------------------------
> Set minimum required R version back to 3.4: as I said elsewhere, R 3.5 is required per Bioconductor policy.

I see! This is somewhat annoying. Guess I will upgrade and reinstall all the packages.

---------------------------------------------
>  I think that Suggests: is the right choice here.

I meant to say Suggests, indeed! This is already the case in the pull request.

---------------------------------------------
> Moreover, splatter is on Bioconductor so there is no need for remotes here.

Fair enough. Right now, splatter and dynverse/splatter contain the same code, so we don't need the remotes here anymore either.

---------------------------------------------
> "Visualizing" to "Visualising": we are using American spelling, so visualizing is correct.

Agreed. Not sure why I changed this.

---------------------------------------------
> Add documentation for curveWeights and pseudotime slots: there are no curveWeights and pseudo time slots in the class.

I see, these were removed around March 2018.

---------------------------------------------
> make sure it passes all checks

It will not, since I configured Travis to treat warnings as errors, but these are not errors caused by me but instead are already present in the package. All of these warnings are documentation-related, but nonetheless important :)

---------------------------------------------
> we are in the middle of a submission to Bioconductor, so we would need to wait until that's done before merging your (updated) pull request.

No problem! Good luck with your submission to Bioconductor (not that this will be a problem)

drisso commented:
Thanks @rcannood !

I agree that we should fix the warnings, and indeed Bioconductor requires no warnings, so I believe @kstreet13 is working on it!

kstreet13 commented:
Thanks @rcannood ! I appreciate you taking the time to put this all together.

Like @drisso said, I'm working on eliminating the warnings, although I think there might be two we can't remove. One is related to the `rgl` package, which is only needed for 3D plotting, and one is for relying on a Fortran function from a different package (`princurve`), which I don't think we can get around. You are correct that the extra 3D plotting code was indeed unnecessary.

rcannood commented:
Good to hear you're working on eliminating the warnings :) I fixed a few more, today. I hope this helps out a little bit.

### Travis ymls
Commit 9f9052d


I also added a `before_script` section to the `.travis.yml`, in order to not get any warnings when rgl decides to wake up on travis. Maybe this is the problem you were referring to earlier? The warning looked like this:
```
checking whether package ‘slingshot’ can be installed ... WARNING
Found the following significant warnings:
  Warning: 'rgl_init' failed, running with rgl.useNULL = TRUE
```

I also included a new file, `.travis_init1.yml`. This is a problem we've had a lot with [dynmethods](https://github.com/dynverse/dynmethods), where installing all of the imports and suggested packages takes such a long time that travis would timeout. Whenever the package cache gets deleted, you can use the init script to submit a custom config to travis (More Options > Trigger Build > Custom Config). The config will only install the packages; skip any examples, vignettes, tests, etc; and then upload the cache to travis. Afterwards, you can submit slingshot commits like before.

### .Rbuildignore
Commit f6810b1

Since I created a `.travis_init1.yml` file, I now got the message:
```
* checking for hidden files and directories ... NOTE
Found the following hidden files and directories:
  .travis_init1.yml
These were most likely included in error. See section ‘Package
structure’ in the ‘Writing R Extensions’ manual.
```
So I edited the .Rbuildignore to include all `^\.travis` files. While I was looking at the file, I noticed that `[.]` was being used instead of `\.` ─ regex-wise, `[.]` is the same as `.`, I think? 

### Build vignettes
Commit 35e2a25

Finally, since the `.travis.yml` contains a `--no-build-vignettes` part, R cmd check will complain that the vignettes in the `vignette/` folder don't have corresponding files in the `inst/doc/` folder:
```
* checking files in ‘vignettes’ ... WARNING
Files in the 'vignettes' directory but no files in 'inst/doc':
  ‘bibFile.bib’, ‘slingshot.Rmd’
```
This was solved by running `devtools::build_vignettes()` locally and committing the output.

### Fortran from princurve
With the fortran function from princurve, I believe you mean this output? 
```
* checking foreign function calls ... NOTE
Foreign function call to a different package:
  .Fortran("getlam", ..., PACKAGE = "princurve")
```

kstreet13 commented:
Hey @rcannood 

Thanks for the help! I just merged your edits into the develop branch (we're holding off on updating master until the package is on Bioconductor).

And yes, that is the output I was referencing. Unfortunately, we need a slightly different version of the get.lam function than the one exported by the princurve package. Their version only returns the sum of the projection distances, but we need each individual projection distance.
